File: testMongodb.py
from pymongo import MongoClient, errors


try:
	'''Connect with local mongdb'''
	# MongoClient = MongoClient('localhost',27017)

	'''Connect with mongdb cluster'''
	MongoClient = MongoClient("mongodb://yfamy123:amy920420@cluster0-shard-00-00-fnmen.mongodb.net:27017,cluster0-shard-00-01-fnmen.mongodb.net:27017,cluster0-shard-00-02-fnmen.mongodb.net:27017/test?ssl=true&replicaSet=Cluster0-shard-0&authSource=admin")

	print("Connect successfully!")
except errors.ConnectionFailure:
   print("Could not connect to MongoDB")

print("Show database", MongoClient.database_names())

'''Change database'''
database = MongoClient.test

'''Get collection name'''
collection = database.collection_names(include_system_collections=False)

for collect in collection:
    print(collect)

'''Get collection restaurants'''
restaurants = database.restaurants

'''Print one sample'''
print(restaurants.find_one())

'''Print one sample with contrain'''
print(restaurants.find_one({ "name": "Wendy'S" }))

File: testMongodb.java
/**
 * Created by fan on 2/4/18.
 */

import com.mongodb.*;

import java.util.List;
import java.util.Set;

public class testMongodb {

    public static void main(String[] args) {

        try {

            /* Connect to Cluster */
            MongoClientURI uri = new MongoClientURI(
                    "mongodb+srv://yfamy123:<password>@cluster0-fnmen.mongodb.net/test");
            MongoClient mongoClient = new MongoClient(uri);

            /* Connect to Local */
//            MongoClient mongoClient = new MongoClient();

            System.out.println("Connect successfully!");

            List<String> databaseNames = mongoClient.getDatabaseNames();

            System.out.println("Database list");
            for(String db : databaseNames) {
                System.out.println(db);
            }

            DB database = mongoClient.getDB("test");

            Set<String> collectionNames = database.getCollectionNames();

            System.out.println("Collection list in database test");
            for(String collection : collectionNames) {
                System.out.println(collection);
            }

            DBCollection restaurant = database.getCollection("restaurants");

            System.out.println("Show one document in the collection");
            System.out.println(restaurant.findOne());

            BasicDBObject whereQuery = new BasicDBObject();
            whereQuery.put("name", "Wendy'S");

            DBCursor cursor = restaurant.find(whereQuery);
            while(cursor.hasNext()) {
                System.out.println(cursor.next());
            }


        } catch (Exception e) {
            System.out.println("Could not connect to MongoDB");
        }

    }



}
